clap_lex/src/ext.rs

```diff
@@ -193,6 +193,7 @@ pub trait OsStrExt: private::Sealed {
     /// assert_eq!("Per", first);
     /// assert_eq!(" Martin-Löf", last);
     /// ```
+    #[deprecated(since = "4.1.0", note = "This is not sound for all `index`")]
     fn split_at(&self, index: usize) -> (&OsStr, &OsStr);
```

Comment:
Shouldn't this one also be marked `unsafe`? Technically a BC break, but that's ok given the time scale and circumstances.

Reply:
I'd rather just soon release a v0.5 with it removed.  This is an intermediate release so it will work with clap v4.1.x (as master is already prepped for v4.2.x)